### PR TITLE
Async/Sync Tests Rewrite

### PR DESCRIPTION
**Pull request by Luc1412**
# Async/Sync Tests Rewrite
Currently, the tests for both the async and sync clients have separate files, and thus, have a lot of repeated code.

I aim to merge this into one, ensuring that both clients are functioning as expected and strengthening the safety of all available tests.

## Overall Idea
I'm planning to create some sort of mock Client/SyncClient hybrid that calls methods from both clients at once when requested. Pytest tests do not run in parallel, so although this is not the best solution it will ensure the following:

- That the Client and SyncClient return the same values, and
- That the Client and SyncClient have all the same functions.

This will effectively merge the two independent test files into one, allowing for easier maintainability and readability. 

This PR stems from the ongoing #29 PR to migrate all new features to this new test system.

## Misc Changes
- Updated `abc.ReconstructAble` to fix changes with `from_dict` classmethod typing annotations.
- Finalized the migration from Python 3.8 to Python 3.9 type annotations. This included the following files: `attribute_table.py`, `examples/discord_integration.py`, `abc.py`, `aes.py`, `asset.py`, `banner.py`, `client.py`, `cosmetics/*.py`, `cosmetics/variants/*.py`, `errors.py`, `http.py`, `images.py`, `map.py`, `new.py`, `new_display_asset.py`, `news.py`, `playlist.py`, `proxies.py`, `shop.py`, and `stats.py`.
- Removed the `synchronize` workflow trigger in the `pull_request` category. This caused tests to run twice in an active PR. Now, tests run when pushed to a branch, when a PR is opened, or when a PR is reopened.

### CODE REVIEW

File: tests/README.md
Comment (Luc1412) [inline tests/README.md:1]: I think it would be nice to directly embed links to those files within the readme for easy access.

File: tests/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+Due to the library's complexity, especially considering that it fully supports both `async` and `sync` functionality,
+many edge cases are tested. Mainly, these tests are related to the `Client` and `SyncClient` classes, and the methods
+defined on them.
+
+#### Definition and Tests for the Hybrid Client
+
+##### Test Client Hybrid: `test_client_hybrid.py`
+
+The tests define a custom `ClientHybrid` class (in `./client/test_client_hybrid.py`). This class wraps a `Client` to act as an intermediatory between a requested API call and the actual method. When an API call is requested, the `ClientHybrid` will call **both** the async `Client` version and the `SyncClient` version of the method. The results are then compared to ensure that they are the same.
```
Comment (Luc1412) [inline tests/README.md]: Any reason for `./` within the brackets?

File: tests/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+Thus, all tests that make API calls will import and use the `ClientHybrid`.
+
+As an example, consider the user requesting to call `fetch_aes()` using the `ClientHybrid`:
+
+- The `ClientHybrid` class is initialized as a context manager, the same as you would with a `Client`.
+- The `fetch_aes()` method is called on the `ClientHybrid`.
+- The sync method of `fetch_aes()` is called on an internally held `SyncClient` class.
+- The async method of `fetch_aes()` is called on the `Client` itself.
+- The result, if reconstructable or comparable, is checked to ensure that both returned objects are the same.
+- The result of the async method call is returned as the final value.
```
Comment (Luc1412) [inline tests/README.md]:
```suggestion
1. The `ClientHybrid` class is initialized as a context manager, the same as you would with a `Client`.
2. The `fetch_aes()` method is called on the `ClientHybrid`.
3. The sync method of `fetch_aes()` is called on an internally held `SyncClient` class.
4. The async method of `fetch_aes()` is called on the `Client` itself.
5. The result, if reconstructable or comparable, is checked to ensure that both returned objects are the same.
6. The result of the async method call is returned as the final value.
```
Think a ordered list might be better here.

**Review by Luc1412 — COMMENTED** [review]
Looks pretty good to me. Addionally we might want to link the Tests README within the normal README. Also the Changelog note needs to be added, since it got lost during re-creating the PR.
